compiler/rustc_ast_pretty/src/pprust/state.rs

```diff
@@ -1709,6 +1709,12 @@ impl<'a> State<'a> {
                     self.print_expr(e, FixupContext::default());
                 }
             }
+            PatKind::Guard(subpat, condition) => {
+                self.print_pat(subpat);
+                self.space();
+                self.word_space("if");
+                self.print_expr(condition, FixupContext::default());
```

Comment:
this might not be right in all contexts as parentheses might be required. Please leave a fixme so we can track that.

Reply:
should I just popen/pclose around it then? better to be correct with too many parens.

Reply:
Seems okay to me, since we're not trying to be rustfmt here